lib/Search/Elasticsearch/Client/0_90/Async/Bulk.pm

```diff
@@ -36,14 +35,13 @@ sub add_action {
     my $weak_add;
     my $add = sub {
         while (@actions) {
-            my @json = try {
+            my @json = eval {
                 $self->_encode_action( splice( @actions, 0, 2 ) );
-            }
-            catch {
-                $self->on_fatal->($_);
-                $deferred->reject($_);
-                ();
             };
+            if ($@) {
```

Comment:
Please do not rely on truthiness of $@ to check for errors (see BACKGROUND in the Try::Tiny docs). Instead this could be written something like:

```
my @json;
unless (eval {
  @json = $self->_encode_action( splice( @actions, 0, 2 ) );
  1;
}) {
  $self->on_fatal->($@);
  $deferred->reject($@);
}

Reply:
I'd add that ideally, it would localize its use of `$@` so as to lessen the chance of confusing poorly-written end user code, but this would require more care to avoid introducing bugs. Handling all this correctly is why Try::Tiny and Syntax::Keyword::Try exist - unfortunately the first is slow and the second requires Perl 5.14.

Reply:
I'm not sure I agree it's required but I've made the changes anyways.